path/posix/resolve.ts

```diff
@@ -16,8 +16,9 @@ export function resolve(...pathSegments: string[]): string {
   for (let i = pathSegments.length - 1; i >= -1 && !resolvedAbsolute; i--) {
     let path: string;
 
-    if (i >= 0) path = pathSegments[i];
-    else {
+    if (i >= 0) {
+      path = pathSegments[i]!;
```

Comment:
`i` starts at `pathSegments.length` and decrements down to -1, however this statement is protected with `i >= 0` so the `i` should always be inbounds.